src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,59 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MAX_AGE = 18;
+    private static final String REGEX_PASSWORD =
+            "(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=\\S+$).{6,}";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (!nullChecker(user).equals("Ok")) {
+            exeptionTrows("nullChecker", nullChecker(user));
+        }
+        userAgeCheck(user.getAge());
+        userLoginCheck(user.getLogin());
+        userPasswordLengthCheck(user.getPassword());
+        storageDao.add(user);
+        return user;
+    }
+
+    private String nullChecker(User user) {
+        return user.getPassword() == null ? "Password"
+                : user.getLogin() == null ? "Login"
+                : user.getLogin().isBlank() ? "Login"
+                : user.getAge() == null ? "Age"
+                : "Ok";
```

Comment:
it would be better to throw exception when something is null, and add `user == null`